service/auth_test.go

```diff
@@ -0,0 +1,72 @@
+package service_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/fabric8-services/fabric8-auth-client/auth"
+	"github.com/fabric8-services/fabric8-common/service"
+	testsuite "github.com/fabric8-services/fabric8-common/test/suite"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	gock "gopkg.in/h2non/gock.v1"
+)
+
+var url = "https://auth.prod-preview.openshift.io"
+
+type AuthServiceTestSuite struct {
+	testsuite.UnitTestSuite
+	authService service.Auth
+}
+
+func TestAuthServiceTestSuite(t *testing.T) {
+	suite.Run(t, &AuthServiceTestSuite{UnitTestSuite: testsuite.NewUnitTestSuite()})
+}
+
+func (s *AuthServiceTestSuite) SetupSuite() {
+	s.UnitTestSuite.SetupSuite()
+
+	var err error
+	s.authService, err = service.NewAuthService(url)
+	require.NoError(s.T(), err)
+}
+
+func (s *AuthServiceTestSuite) TestCheckSpaceScope() {
+	s.T().Run("scope_found_true", func(t *testing.T) {
```

Comment:
We have to check the request as well. Is the request to the Auth service has all expected headers? Authorization header has expected token? Request ID header is present? URL is expected?
This should be checked for all requests in the test.

Also check all potential responses from Auth: https://github.com/fabric8-services/fabric8-auth/blob/48c0a1b90a0ff1b7cd8621c67acd8c5c13a4cb3b/design/resource.go#L68-L71
Which are: 200, 401, 500, 404 (when the resource ID is unknown)

Reply:
@alexeykazakov plz check logic [code_link](https://github.com/fabric8-services/fabric8-common/pull/62/files#diff-0866f3e18030ce6c08767bc62b9b8011R55) for anything other than 200, it throws exception.  We already have 401 handle in test [test_code_link](https://github.com/fabric8-services/fabric8-common/pull/62/files#diff-678018887df97d05396e5b0104a16d66R66) so I guess no more test needed.

Reply:
How do you know if throws an error if it's other than 200 if you don't have tests for that?
Please add the tests I'm asking for.
Also please add checks for the requests:

> We have to check the request as well. Is the request to the Auth service has all expected headers? Authorization header has expected token? Request ID header is present? URL is expected?
This should be checked for all requests in the test.

Reply:
Done in https://github.com/fabric8-services/fabric8-common/pull/62/commits/ebc712d2d9aac03e2701a07423c59c3999224d3b